src/main/java/com/epam/spring/homework3/repository/impl/FavorRepositoryImpl.java

```diff
@@ -0,0 +1,104 @@
+package com.epam.spring.homework3.repository.impl;
+
+import com.epam.spring.homework3.model.Favor;
+import com.epam.spring.homework3.model.enums.Speciality;
+import com.epam.spring.homework3.repository.FavorRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+@Slf4j
+@Component
```

Comment:
Use @Repository annotation instead of @Component, it has actually no difference, but first one has more readability to reviewers